tools/push-multiarch

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -ex
+
+IMAGE_NAME=${1}
+
+WORKSPACE_STATUS=`tools/get_workspace_status.sh`
+
+STABLE_DOCKER_REGISTRY=`echo "${WORKSPACE_STATUS}" | grep STABLE_DOCKER_REGISTRY | awk '{print $2}'`
+STABLE_DOCKER_IMAGE_PREFIX=`echo "${WORKSPACE_STATUS}" | grep STABLE_DOCKER_IMAGE_PREFIX | awk '{print $2}'`
+STABLE_DOCKER_TAG=`echo "${WORKSPACE_STATUS}" | grep STABLE_DOCKER_TAG | awk '{print $2}'`
+
+# Needed for manifest commands
+export DOCKER_CLI_EXPERIMENTAL=enabled
+
+IMAGE_BASE="${STABLE_DOCKER_REGISTRY}/${STABLE_DOCKER_IMAGE_PREFIX}${IMAGE_NAME}"
+
+MULTIARCH="${STABLE_DOCKER_REGISTRY}/${STABLE_DOCKER_IMAGE_PREFIX}${IMAGE_NAME}:${STABLE_DOCKER_TAG}"
+ARCH_AMD64="${IMAGE_BASE}-amd64:${STABLE_DOCKER_TAG}"
+ARCH_ARM64="${IMAGE_BASE}-arm64:${STABLE_DOCKER_TAG}"
+
+MULTIARCH=justinsb/etcdmanager-multiarch:latest
+
+docker pull ${ARCH_AMD64}
+docker pull ${ARCH_ARM64}
+
+#docker tag ${IMAGE_BASE}-amd64:${STABLE_DOCKER_TAG} ${IMAGE_BASE}:${STABLE_DOCKER_TAG}-amd64
+#docker push ${IMAGE_BASE}:${STABLE_DOCKER_TAG}-amd64
+#docker tag ${IMAGE_BASE}-arm64:${STABLE_DOCKER_TAG} ${IMAGE_BASE}:${STABLE_DOCKER_TAG}-arm64
+#docker push ${IMAGE_BASE}:${STABLE_DOCKER_TAG}-arm64
+
+docker manifest create ${MULTIARCH} ${ARCH_AMD64} ${ARCH_ARM64}
+docker manifest annotate ${MULTIARCH} ${ARCH_AMD64} --os linux --arch amd64
+docker manifest annotate ${MULTIARCH} ${ARCH_ARM64} --os linux --arch arm64
+
+
+#docker manifest create --amend ${IMAGE_BASE}:${STABLE_DOCKER_TAG} ${IMAGE_BASE}:${STABLE_DOCKER_TAG}-amd64 ${IMAGE_BASE}:${STABLE_DOCKER_TAG}-arm64
```

Comment:
This is enough in case of GCP, which allows layers to be shared between repos.

DockerHub seems to have disabled manifest with layers form other repos (or just another bug). If you want to continue using DH for now, maybe we should just push to `etcd-manager:$arch-$tag` for now and keep the manifest pushing simple.

Reply:
A quick proposal on how to to that: https://github.com/kopeio/etcd-manager/pull/334/commits/15b0615832cece27c4a7f1c08f3873d8fa384bc6